datafusion/core/src/datasource/physical_plan/parquet/page_filter.rs

```diff
@@ -236,6 +225,24 @@ impl PagePruningPredicate {
     }
 }
 
+/// returns the number of rows skipped in the selection
+/// TODO should this be upstreamed to RowSelection?
+fn rows_skipped(selection: &RowSelection) -> usize {
+    selection
+        .iter()
+        .fold(0, |acc, x| if x.skip { acc + x.row_count } else { acc })
+}
+
+fn update_selection(
+    current_selection: Option<RowSelection>,
+    row_selection: RowSelection,
+) -> Option<RowSelection> {
+    match current_selection {
+        None => Some(row_selection),
+        Some(current_selection) => Some(current_selection.intersection(&row_selection)),
```

Comment:
@alamb  maybe we can check after the `intersection` the result  row_selection is select 0 row ? 🤔  then we can cast it to skip. IMO select-0 seems need do more effort in next round check.

Reply:
Good idea. Thank you @Ted-Jiang I made this change in e2dec6120